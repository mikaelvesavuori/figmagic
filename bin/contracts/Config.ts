export type Config = {
  debugMode: boolean;
  camelizeTokenNames: boolean;
  borderWidthUnit: BorderWidthUnit;
  fontUnit: FontUnits;
  letterSpacingUnit: LetterSpacingUnit;
  lineHeightUnit: LineHeightUnit;
  opacitiesUnit: OpacitiesUnit;
  radiusUnit: RadiusUnit;
  shadowUnit: ShadowUnit;
  durationUnit: DurationUnit;
  spacingUnit: SpacingUnit;
  figmaData: string;
  figmagicFolder: string;
  outputFolderElements: string;
  outputFolderGraphics: string;
  outputFolderTokens: string;
  outputFormatColors: OutputFormatColors;
  outputFormatCss: OutputFormatCss;
  outputFormatDescription: OutputFormatDescription;
  outputFormatElements: OutputFormatElements;
  outputFormatGraphics: OutputFormatGraphics;
  outputFormatStorybook: OutputFormatStorybook;
  outputFormatTokens: OutputFormatTokens;
  outputGraphicElements: boolean;
  outputGraphicElementsMap: boolean;
  outputScaleGraphics: number;
  outputDataTypeToken: OutputDataTypeToken;
  overwrite: Overwrite;
  recompileLocal: boolean;
  remSize: number;
  skipFileGeneration: {
    forceUpdate: boolean;
    skipCss: boolean;
    skipDescription: boolean;
    skipReact: boolean;
    skipStorybook: boolean;
    skipStyled: boolean;
  };
  syncElements: boolean;
  syncGraphics: boolean;
  syncTokens: boolean;
  templates: {
    templatePathGraphic: string;
    templatePathReact: string;
    templatePathStorybook: string;
    templatePathStyled: string;
  };
  token: string | null;
  tokensRelativeImportPrefix: string;
  unitlessPrecision: number;
  url: string | null;
  usePostscriptFontNames: boolean;
  useLiteralFontFamilies: boolean;
  versionName: string | null;
};

export type ConfigDTO = {
  debugMode?: boolean;
  camelizeTokenNames?: boolean;
  borderWidthUnit?: BorderWidthUnit;
  fontUnit?: FontUnits;
  letterSpacingUnit?: LetterSpacingUnit;
  lineHeightUnit?: LineHeightUnit;
  opacitiesUnit?: OpacitiesUnit;
  radiusUnit?: RadiusUnit;
  shadowUnit?: ShadowUnit;
  durationUnit?: DurationUnit;
  spacingUnit?: SpacingUnit;
  figmaData?: string;
  figmagicFolder?: string;
  outputFolderElements?: string;
  outputFolderGraphics?: string;
  outputFolderTokens?: string;
  outputFormatColors?: OutputFormatColors;
  outputFormatCss?: OutputFormatCss;
  outputFormatDescription?: OutputFormatDescription;
  outputFormatElements?: OutputFormatElements;
  outputFormatGraphics?: OutputFormatGraphics;
  outputFormatStorybook?: OutputFormatStorybook;
  outputFormatTokens?: OutputFormatTokens;
  outputGraphicElements?: boolean;
  outputGraphicElementsMap?: boolean;
  outputScaleGraphics?: number;
  outputDataTypeToken?: OutputDataTypeToken;
  overwrite?: {
    css: boolean;
    description: boolean;
    graphic: boolean;
    react: boolean;
    storybook: boolean;
    styled: boolean;
  };
  recompileLocal?: boolean;
  remSize?: number;
  skipFileGeneration?: {
    forceUpdate?: boolean;
    skipCss?: boolean;
    skipDescription?: boolean;
    skipReact?: boolean;
    skipStorybook?: boolean;
    skipStyled?: boolean;
  };
  syncElements?: boolean;
  syncGraphics?: boolean;
  syncTokens?: boolean;
  templates?: {
    templatePathGraphic?: string;
    templatePathReact?: string;
    templatePathStorybook?: string;
    templatePathStyled?: string;
  };
  token?: string | null;
  tokensRelativeImportPrefix?: string | null;
  unitlessPrecision?: number;
  url?: string | null;
  usePostscriptFontNames?: boolean;
  useLiteralFontFamilies?: boolean;
  versionName?: string | null;
};

export type Overwrite = {
  css: boolean;
  description: boolean;
  graphic: boolean;
  react: boolean;
  storybook: boolean;
  styled: boolean;
};

export type BorderWidthUnit = 'rem' | 'em' | 'px';
export type ShadowUnit = 'rem' | 'em' | 'px';
export type DurationUnit = 's' | 'ms';
export type RadiusUnit = 'rem' | 'em' | 'px';
export type FontUnits = 'rem' | 'em' | 'px';
export type LetterSpacingUnit = 'em' | 'px';
export type LineHeightUnit = 'unitless' | 'px' | 'rem' | 'em';
export type OpacitiesUnit = 'float' | 'percent';
export type SpacingUnit = 'rem' | 'em' | 'px';

export type OutputFormatColors = 'hex' | 'rgba';
export type OutputFormatCss = 'ts' | 'mjs' | 'js';
export type OutputFormatDescription = 'md' | 'txt';
export type OutputFormatElements = 'tsx' | 'jsx';
export type OutputFormatGraphics = 'svg' | 'png';
export type OutputFormatStorybook = 'ts' | 'js' | 'mdx';
export type OutputFormatTokens = 'ts' | 'mjs' | 'js' | 'json';
export type OutputDataTypeToken = null | 'enum';

export type TemplatesConfig = {
  templates: {
    templatePathGraphic: string;
    templatePathReact: string;
    templatePathStorybook: string;
    templatePathStyled: string;
  };
};
