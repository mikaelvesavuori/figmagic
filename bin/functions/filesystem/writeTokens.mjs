import { camelize } from '../helpers/camelize.mjs';
import { processTokens } from '../process/processTokens.mjs';
import { writeFile } from './writeFile.mjs';

import { errorWriteTokens, errorWriteTokensNoSettings } from '../../meta/errors.mjs';
import { acceptedTokenTypes } from '../../meta/acceptedTokenTypes.mjs';

/**
 * Write tokens to file
 *
 * @exports
 * @async
 * @function
 * @param {array} tokens - The final array of design tokens
 * @param {object} config - User configuration object
 * @returns {boolean} - Returns true when finished
 * @throws {errorWriteTokens} - Throws error when no tokens are provided
 * @throws {errorWriteTokens} - Throws error when tokens are zero-length
 * @throws {errorWriteTokensNoSettings} - Throws error when missing config
 */
export async function writeTokens(tokens, config) {
  if (!tokens) throw new Error(errorWriteTokens);
  if (!(tokens.length > 0)) throw new Error(errorWriteTokens);
  if (!config) throw new Error(errorWriteTokensNoSettings);

  const tokensToProcess = new Promise((resolve, reject) => {
    try {
      tokens.forEach(async (token) => {
        const tokenName = camelize(token.name);

        if (acceptedTokenTypes.includes(tokenName.toLowerCase())) {
          const PROCESSED_TOKEN = processTokens(token, tokenName, config);

          if (config.debugMode) console.log(PROCESSED_TOKEN);

          await writeFile(
            PROCESSED_TOKEN,
            config.outputFolderTokens,
            tokenName,
            'token',
            config.outputTokenFormat,
            { dataType: config.outputTokenDataType }
          );
        }
      });

      resolve(true);
    } catch (error) {
      reject(error);
    }
  });

  return tokensToProcess.catch((error) => console.error(error));
}
