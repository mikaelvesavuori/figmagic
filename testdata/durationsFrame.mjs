export const durationsFrame = {
  id: '2657:97',
  name: 'Durations',
  type: 'FRAME',
  blendMode: 'PASS_THROUGH',
  children: [
    {
      id: '63:0',
      name: 'Decimal',
      type: 'TEXT',
      blendMode: 'PASS_THROUGH',
      absoluteBoundingBox: [Object],
      constraints: [Object],
      fills: [Array],
      strokes: [],
      strokeWeight: 1,
      strokeAlign: 'OUTSIDE',
      exportSettings: [Array],
      effects: [],
      characters: '0.5',
      style: [Object],
      layoutVersion: 1,
      characterStyleOverrides: [],
      styleOverrideTable: {}
    },
    {
      id: '2657:100',
      name: 'Fast',
      type: 'TEXT',
      blendMode: 'PASS_THROUGH',
      absoluteBoundingBox: [Object],
      constraints: [Object],
      fills: [Array],
      strokes: [],
      strokeWeight: 1,
      strokeAlign: 'OUTSIDE',
      exportSettings: [Array],
      effects: [],
      characters: '200',
      style: [Object],
      layoutVersion: 1,
      characterStyleOverrides: [],
      styleOverrideTable: {}
    },
    {
      id: '2657:101',
      name: 'Medium',
      type: 'TEXT',
      blendMode: 'PASS_THROUGH',
      absoluteBoundingBox: [Object],
      constraints: [Object],
      fills: [Array],
      strokes: [],
      strokeWeight: 1,
      strokeAlign: 'OUTSIDE',
      exportSettings: [Array],
      effects: [],
      characters: '400',
      style: [Object],
      layoutVersion: 1,
      characterStyleOverrides: [],
      styleOverrideTable: {}
    },
    {
      id: '2657:99',
      name: 'Slow',
      type: 'TEXT',
      blendMode: 'PASS_THROUGH',
      absoluteBoundingBox: [Object],
      constraints: [Object],
      fills: [Array],
      strokes: [],
      strokeWeight: 1,
      strokeAlign: 'OUTSIDE',
      exportSettings: [Array],
      effects: [],
      characters: '750',
      style: [Object],
      layoutVersion: 1,
      characterStyleOverrides: [],
      styleOverrideTable: {}
    }
  ],
  absoluteBoundingBox: { x: 1152, y: 931, width: 593, height: 75 },
  constraints: { vertical: 'TOP', horizontal: 'LEFT' },
  clipsContent: true,
  background: [{ blendMode: 'NORMAL', type: 'SOLID', color: [Array] }],
  fills: [{ blendMode: 'NORMAL', type: 'SOLID', color: [Array] }],
  strokes: [],
  strokeWeight: 1,
  strokeAlign: 'INSIDE',
  backgroundColor: { r: 1, g: 1, b: 1, a: 1 },
  effects: []
};
