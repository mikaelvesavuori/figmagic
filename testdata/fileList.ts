export const fileList = [
  {
    url: 'https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png',
    file: 'example.png'
  },
  {
    url: 'https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg',
    file: 'example.svg'
  }
];

export const fileListSVG = [
  {
    url: 'https://example.com',
    file: 'check.svg'
  },
  {
    url: 'https://example.com',
    file: 'recycle.svg'
  },
  {
    url: 'https://example.com',
    file: 'close.svg'
  },
  {
    url: 'https://example.com',
    file: 'more.svg'
  }
];
