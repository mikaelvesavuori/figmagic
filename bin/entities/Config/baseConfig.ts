import { Config } from '../../contracts/Config';

export const baseConfig: Config = {
  debugMode: false,
  camelizeTokenNames: true,
  borderWidthUnit: 'px',
  fontUnit: 'rem',
  letterSpacingUnit: 'em',
  lineHeightUnit: 'unitless',
  opacitiesUnit: 'float',
  radiusUnit: 'px',
  shadowUnit: 'px',
  spacingUnit: 'rem',
  durationUnit: 's',
  figmaData: 'figma.json',
  figmagicFolder: '.figmagic',
  outputFolderElements: 'elements',
  outputFolderGraphics: 'graphics',
  outputFolderTokens: 'tokens',
  outputFormatColors: 'rgba',
  outputFormatCss: 'ts',
  outputFormatDescription: 'md',
  outputFormatElements: 'tsx',
  outputFormatGraphics: 'svg',
  outputFormatStorybook: 'js',
  outputFormatTokens: 'ts',
  outputGraphicElements: false,
  outputGraphicElementsMap: false,
  outputScaleGraphics: 1,
  outputDataTypeToken: null,
  overwrite: {
    css: false,
    description: false,
    graphic: false,
    react: false,
    storybook: false,
    styled: false
  },
  recompileLocal: false,
  remSize: 16,
  skipFileGeneration: {
    forceUpdate: true,
    skipCss: false,
    skipDescription: false,
    skipReact: false,
    skipStorybook: false,
    skipStyled: false
  },
  syncElements: false,
  syncGraphics: false,
  syncTokens: true,
  templates: {
    templatePathGraphic: './node_modules/figmagic/templates/graphic',
    templatePathReact: './node_modules/figmagic/templates/react',
    templatePathStorybook: './node_modules/figmagic/templates/story',
    templatePathStyled: './node_modules/figmagic/templates/styled'
  },
  token: '',
  tokensRelativeImportPrefix: '',
  unitlessPrecision: 2,
  url: '',
  usePostscriptFontNames: false,
  useLiteralFontFamilies: false,
  versionName: null
};
