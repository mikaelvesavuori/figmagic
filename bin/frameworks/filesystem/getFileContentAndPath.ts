import { GetFileDataOperation, FileContentWithPath } from '../../contracts/Write';
import {
  PrepComponent,
  PrepCss,
  PrepDescription,
  PrepStorybook,
  PrepStyledComponents,
  PrepGraphicComponent
} from '../../contracts/PrepFile';
import { ProcessedToken } from '../../contracts/ProcessedToken';

import {
  prepComponent,
  prepStyledComponents,
  prepCss,
  prepStorybook,
  prepDescription,
  prepGraphicComponent
} from './prepFile';

import { createEnumStringOutOfObject } from '../../frameworks/string/createEnumStringOutOfObject';

import { MsgGeneratedFileWarning } from '../messages/messages';
import {
  ErrorGetFileContentAndPath,
  ErrorGetFileContentAndPathMissingFields,
  ErrorGetFileContentAndPathNoReturn
} from '../errors/errors';

/**
 * Orchestrator to get file content and path, before writing files
 */
export function getFileContentAndPath(
  getFileContentAndPathOperation: GetFileDataOperation
): FileContentWithPath {
  if (!getFileContentAndPathOperation) throw Error(ErrorGetFileContentAndPath);
  if (!checkIfFieldsExist(getFileContentAndPathOperation))
    throw Error(ErrorGetFileContentAndPathMissingFields);

  const {
    type,
    file,
    path,
    name,
    format,
    text,
    element,
    imports,
    extraProps,
    metadata,
    templates
  } = getFileContentAndPathOperation;

  let filePath = `${path}/${name}`;

  const fileOperations = {
    raw: () => {
      return { fileContent: `${JSON.stringify(file, null, ' ')}`, filePath };
    },
    token: () => {
      if (metadata?.dataType === 'enum')
        return { fileContent: getTokenString(file, name, format, metadata.dataType), filePath };
      filePath += `.${format}`;
      return { fileContent: getTokenString(file, name, format), filePath };
    },
    component: () => {
      if (type === 'component' && templates)
        return prepComponent({
          name,
          filePath,
          format,
          templates,
          text,
          extraProps,
          element
        } as PrepComponent);
    },
    styled: () => {
      if (type === 'styled' && templates)
        return prepStyledComponents({
          name,
          filePath,
          format,
          templates,
          element
        } as PrepStyledComponents);
    },
    story: () => {
      if (type === 'story' && templates)
        return prepStorybook({ name, filePath, format, templates, text } as PrepStorybook);
    },
    css: () => prepCss({ name, filePath, format, imports, file } as PrepCss),
    description: () => prepDescription({ filePath, file, format } as PrepDescription),
    graphic: () =>
      prepGraphicComponent({
        name,
        filePath,
        format,
        templates,
        file
      } as PrepGraphicComponent)
  };

  // @ts-ignore
  if (fileOperations.hasOwnProperty(type)) return fileOperations[type]();
  else throw Error(ErrorGetFileContentAndPathNoReturn);
}

/**
 * @description Get file data string for tokens
 */
const getTokenString = (
  file: string | ProcessedToken,
  name: string,
  format: string,
  dataType?: string
): string => {
  const exportString = format === 'js' ? `module.exports = ${name}` : `export default ${name}`;
  const constAssertion = format === 'ts' ? ' as const;' : '';

  if (format === 'json') return getTokenStringJSON(file);
  if (format === 'css') return getTokenStringCSS(file, name);
  if (dataType === 'enum') return getTokenStringEnum(file, name, exportString);
  return getTokenStringJS(file, name, exportString, constAssertion);
};

/**
 * @description Return JSON token string
 */
function getTokenStringJSON(file: string | ProcessedToken) {
  return `${JSON.stringify(file, null, ' ')}`;
}

/**
 * @description Return CSS variables token string
 */
function getTokenStringCSS(file: string | ProcessedToken, name: string) {
  const contents: any = file;
  let css = ':root {\n';

  for (const key in contents) {
    const value = contents[key];
    css += `  --${name}-${key}: ${value};\n`;
  }

  css += '}\n';

  return css;
}

/**
 * @description Return enum token string
 */
function getTokenStringEnum(file: string | ProcessedToken, name: string, exportString: string) {
  return `// ${MsgGeneratedFileWarning}\n\nenum ${name} {${createEnumStringOutOfObject(
    file
  )}\n}\n\n${exportString};`;
}

/**
 * @description Return TS/JS token string
 */
function getTokenStringJS(
  file: string | ProcessedToken,
  name: string,
  exportString: string,
  constAssertion: string
) {
  return `// ${MsgGeneratedFileWarning}\n\nconst ${name} = ${JSON.stringify(
    file,
    null,
    ' '
  )}${constAssertion}\n\n${exportString};`;
}

/**
 * @description Validate whether required fields exist on GetFileDataOperation type
 */
const checkIfFieldsExist = (getFileContentAndPathOperation: GetFileDataOperation): boolean => {
  if (
    !getFileContentAndPathOperation.type ||
    !getFileContentAndPathOperation.file ||
    !getFileContentAndPathOperation.path ||
    !getFileContentAndPathOperation.name ||
    !getFileContentAndPathOperation.format ||
    !getFileContentAndPathOperation.element
  )
    return false;
  return true;
};
