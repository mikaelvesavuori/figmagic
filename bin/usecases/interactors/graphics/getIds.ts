import { FRAME as Frame } from '../../../contracts/Figma';

import { ErrorGetIds, ErrorGetIdsComponents } from '../../../frameworks/errors/errors';

/**
 * @description Get IDs from graphics page
 */
export const getIds = (graphicsPage: Frame[]): Record<string, unknown>[] => {
  if (!graphicsPage) throw Error(ErrorGetIds);
  if (!(graphicsPage.length > 0)) throw Error(ErrorGetIds);

  const ids = graphicsPage
    .filter((item: any) => item.type === 'COMPONENT')
    .map((item: any) => {
      return { id: item.id, name: item.name };
    });

  if (!(ids.length > 0)) throw Error(ErrorGetIdsComponents);

  return ids;
};
