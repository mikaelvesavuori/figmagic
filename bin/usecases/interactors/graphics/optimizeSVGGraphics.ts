import { optimize } from 'svgo';
import { Config } from '../../../contracts/Config';
import { FileList } from '../../../contracts/Files';
import {
  ErrorOptimizeSVGGraphics,
  ErrorOptimizeSVGGraphicsConfig,
  ErrorOptimizeSVGGraphicsFormat
} from '../../../frameworks/errors/errors';
import { loadFile } from '../../../frameworks/filesystem/loadFile';
import { write } from '../../../frameworks/filesystem/write';

/**
 * @description Optimize SVG Graphics downloaded from Figma using svgo
 */
export async function optimizeSVGGraphics(fileList: FileList[], config: Config): Promise<void> {
  if (!fileList || !config) throw Error(ErrorOptimizeSVGGraphicsConfig);

  const { outputFormatGraphics, outputFolderGraphics } = config;

  if (outputFormatGraphics !== 'svg') throw Error(ErrorOptimizeSVGGraphicsFormat);

  await Promise.all(
    fileList.map(async (file: FileList) => {
      return new Promise(async (resolve) => {
        const svg = loadFile(`${outputFolderGraphics}/${file.file}`);
        if (typeof svg !== 'string') {
          throw Error(ErrorOptimizeSVGGraphics(`Invalid SVG file: ${file.file}`));
        }
        const result = optimize(svg, {
          multipass: true,
          plugins: [
            'preset-default',
            {
              name: 'removeViewBox',
              active: false
            },
            {
              name: 'reusePaths',
              active: true
            }
          ],
          js2svg: {
            indent: 2,
            pretty: true
          }
        });
        if (result.error !== undefined) {
          throw Error(ErrorOptimizeSVGGraphics(result.error));
        }
        const optimizedSvgString = result.data;
        await write(`${outputFolderGraphics}/${file.file}`, optimizedSvgString);
        resolve(true);
      });
    })
  );
}
