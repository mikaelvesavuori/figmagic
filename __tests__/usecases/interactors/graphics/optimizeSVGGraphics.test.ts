import fs from 'fs';
import trash from 'trash';
import { baseConfig } from '../../../../bin/entities/Config/baseConfig';
import { loadFile } from '../../../../bin/frameworks/filesystem/loadFile';
import { refresh } from '../../../../bin/frameworks/filesystem/refresh';
import { optimizeSVGGraphics } from '../../../../bin/usecases/interactors/graphics/optimizeSVGGraphics';
import { fileListSVG } from '../../../../testdata/fileList';

// Set temp folder
const TEMP_FOLDER = __dirname + `/__graphics__`;
const TEST_DATA_SVG_FOLDER = __dirname + '/../../../../testdata/svg';

describe('Failure cases', () => {
  test('It should throw an error if no argument is provided', async () => {
    // @ts-ignore
    await expect(optimizeSVGGraphics()).rejects.toThrow();
  });
  test('It should throw an error if graphics output format is not SVG', async () => {
    await expect(
      optimizeSVGGraphics(fileListSVG, {
        ...baseConfig,
        optimizeSVG: true,
        outputFolderGraphics: TEMP_FOLDER,
        outputFormatGraphics: 'png'
      })
    ).rejects.toThrow();
  });
});

describe('Success cases', () => {
  test('It should successfully optimize SVG graphics if provided valid file list and config', async () => {
    await optimizeSVGGraphics(fileListSVG, {
      ...baseConfig,
      optimizeSVG: true,
      outputFolderGraphics: TEMP_FOLDER,
      outputFormatGraphics: 'svg'
    });

    fileListSVG.forEach(({ file }) => {
      expect(loadFile(`${TEMP_FOLDER}/${file}`)).toMatchSnapshot();
    });
  });
});

beforeAll(async () => {
  refresh(TEMP_FOLDER);
  fileListSVG.forEach(({ file }) => {
    fs.copyFileSync(`${TEST_DATA_SVG_FOLDER}/${file}`, `${TEMP_FOLDER}/${file}`);
  });
});

afterAll(async () => {
  await trash(TEMP_FOLDER);
});
