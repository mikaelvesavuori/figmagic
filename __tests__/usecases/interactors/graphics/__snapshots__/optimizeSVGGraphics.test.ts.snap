// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success cases It should successfully optimize SVG graphics if provided valid file list and config 1`] = `
"<svg width="55" height="56" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M2.111 28.066 17.033 51.5l35.14-50.061" stroke="#219653" stroke-width="5"/>
</svg>
"
`;

exports[`Success cases It should successfully optimize SVG graphics if provided valid file list and config 2`] = `
"<svg width="96" height="96" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="m56.19 41.64-7.74-13.4-7.74 13.4-15.8-9.13 11.2-19.4A8.28 8.28 0 0 1 43.24 9h10.42a8.28 8.28 0 0 1 7.13 4.11L72 32.51l-15.81 9.13Zm-7.74-17.39 8.47 14.65 12.33-7.12-10.2-17.67A6.239 6.239 0 0 0 53.66 11H43.24a6.25 6.25 0 0 0-5.4 3.11L27.65 31.78 40 38.9l8.45-14.65Z" fill="#000"/>
  <path d="M48.45 27.24h-.1a1 1 0 0 1-.9-1.1c0-.39 1.13-9.7 12.14-13.47a1.002 1.002 0 0 1 1.357 1.004 1 1 0 0 1-.707.886c-9.82 3.37-10.79 11.7-10.8 11.79a1 1 0 0 1-.99.89ZM69.65 45l-18.31-3.3 4.77-2.33 14-8.08 5-3.36L69.65 45Zm-12-4.16 10.63 1.89 3.18-9.9-.34.23-13.47 7.78Zm20.46 45.01-22.41-.06V67.54h15.47l-7.7-13.42 15.83-9.07 11.19 19.46a8.26 8.26 0 0 1 0 8.23l-5.23 9a8.25 8.25 0 0 1-7.15 4.11ZM57.7 83.79l20.4.06a6.25 6.25 0 0 0 5.39-3.1l5.23-9a6.23 6.23 0 0 0 0-6.22L78.6 47.82 66.24 54.9l8.42 14.69H57.73l-.03 14.2Z" fill="#000"/>
  <path d="M78.09 85.85h-.19a1 1 0 0 1-.79-1.18c2-10.18-4.7-15.21-4.76-15.26a1 1 0 1 1 1.17-1.61c.32.23 7.82 5.84 5.55 17.26a1 1 0 0 1-.98.79ZM58.08 90.8l-12-13.28 12-14.2-.39 5.3v16.14l.39 6.04Zm-9.32-13.32 7 7.72V69.24l-7 8.24Zm-8.58 8.59L17.79 86a8.25 8.25 0 0 1-7.11-4.15l-5.17-9a8.24 8.24 0 0 1 0-8.22l11.31-19.42 15.77 9.18-7.79 13.37 15.47.07-.09 18.24ZM17.54 48 7.27 65.57a6.23 6.23 0 0 0 0 6.22l5.18 9A6.239 6.239 0 0 0 17.8 84l20.39.08.07-14.24-16.93-.08 8.52-14.63L17.54 48Z" fill="#000"/>
  <path d="M17.86 70.68a18.05 18.05 0 0 1-12.11-4.87 1 1 0 0 1 1.32-1.5c7.75 6.82 15.28 3.66 15.59 3.52a1.005 1.005 0 0 1 .81 1.84 15.18 15.18 0 0 1-5.61 1.01Zm17.19-12.11-4.39-3-13.95-8.11-5.42-2.7 17.5-3.7 6.26 17.51Zm-17.73-13 .37.18 13.43 7.83-3.63-10.2-10.17 2.19Z" fill="#000"/>
</svg>
"
`;

exports[`Success cases It should successfully optimize SVG graphics if provided valid file list and config 3`] = `
"<svg width="54" height="54" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="m1.783 1.783 50.434 50.434m-50.434 0L52.217 1.783" stroke="#000" stroke-width="5"/>
</svg>
"
`;

exports[`Success cases It should successfully optimize SVG graphics if provided valid file list and config 4`] = `
"<svg width="50" height="50" fill="none" xmlns="http://www.w3.org/2000/svg">
  <circle cx="16" cy="25" r="3" fill="#000"/>
  <circle cx="25" cy="25" r="3" fill="#000"/>
  <circle cx="34" cy="25" r="3" fill="#000"/>
  <circle cx="25" cy="25" r="23.5" stroke="#333" stroke-width="3"/>
</svg>
"
`;
